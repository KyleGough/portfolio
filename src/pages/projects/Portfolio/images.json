[
  {
    "src": "/img/portfolio-projects-v1a.jpg",
    "caption": "Portfolio project page version 1"
  },
  {
    "src": "/img/portfolio-projects-v1b.jpg",
    "caption": "Portfolio about page version 1"
  },
  {
    "src": "/img/portfolio-projects-v2.jpg",
    "caption": "Portfolio project page version 2"
  },
  {
    "src": "/img/portfolio-projects-v3.jpg",
    "caption": "Portfolio project page version 3"
  },
  {
    "src": "/img/portfolio-projects-v4a.jpg",
    "caption": "Portfolio project page upper version 4"
  },
  {
    "src": "/img/portfolio-projects-v4b.jpg",
    "caption": "Portfolio project page lower version 4"
  }
]
