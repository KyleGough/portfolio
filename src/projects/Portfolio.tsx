import { useEffect } from 'react';
import Section from '../components/Section';
import Divider from '../components/Divider';
import Chip from '../components/Chip';
import Pagination from '../components/Pagination';
import ImageFigure from '../components/ImageFigure';
import Screenshots from '../components/Screenshots';

export default function Portfolio() {

    useEffect(() => {
        document.title = 'Portfolio - Personal Portfolio Website';
    }, []);

    const images = [
        { src:'/img/portfolio-projects-v1a.jpg', caption:'Portfolio project page version 1' },
        { src:'/img/portfolio-projects-v1b.jpg', caption:'Portfolio about page version 1' },
        { src:'/img/portfolio-projects-v2.jpg', caption:'Portfolio project page version 2' },
        { src:'/img/portfolio-projects-v3.jpg', caption:'Portfolio project page version 3' },
        { src:'/img/portfolio-projects-v4a.png', caption:'Portfolio project page upper version 4' },
        { src:'/img/portfolio-projects-v4b.png', caption:'Portfolio project page lower version 4' }
    ];

    return (
        <>
        <Section>
            <h1 className='project-title'>Portfolio</h1>
            <p className='text-link-hover my-4'><time dateTime='2016-12'>December 2016</time> - <time dateTime='2022-05'>March 2022</time></p>
            <p className='mb-4'>Personal portfolio website created to showcase my projects and technical skills. First created in 2016 as a simple static HTML website, the website has experienced multiple improvements over the years including migration to PHP, then to React with Materialize, and most recently to React with Tailwind.'</p>

            <div className='flex flex-row flex-wrap items-center mt-8'>
                <Chip name='React' />
                <Chip name='JavaScript' />
                <Chip name='TypeScript' />
                <Chip name='Tailwind' />
                <Chip name='HTML' />
                <Chip name='CSS' />
                <Chip name='Node' />
                <Chip disabled name='PHP' />
                <Chip disabled name='Materialize' />
                <Chip disabled name='Bootstrap' />
                <Chip disabled name='Sass' />
            </div>
        </Section>

        <Divider />

        <ImageFigure image={images[4]} />

        <Divider />

        <Section>
            <h2 className='project-header'>Versions</h2>
            <h3 className='mb-4'><strong>Version 1 (HTML)</strong></h3>
            <p className='mb-8'>The initial version created in <time dateTime='2016-12'>December 2016</time> and was a simple site created with HTML, Sass and Bootstrap. As additional project pages were added the CSS framework was ported to Materialize. This version was never made public.</p>

            <h3 className='mb-4'><strong>Version 2 (PHP)</strong></h3>
            <p className='mb-8'>The second iteration of the site was rebuilt in PHP to aid maintainability and expand functionality. Using PHP allowed reusable components such as pagination, navigation bars, footers, and skill progress bars. However these components were not easily maintainable or readable. This version's repository can be viewed
                <a href='https://github.com/KyleGough/portfolio' target='_blank' rel='noopener noreferrer'> here</a>.
            </p>

            <h3 className='mb-4'><strong>Version 3 (React with Materialize)</strong></h3>
            <p className='mb-8'>The third iteration was rebuilt again in <time dateTime='2020-05'>May 2020</time> with React due to its benefits and ease of use over PHP. After using React in multiple other projects, I decided to rebuild the site taking effort to replicate most of the previous styling but making specific changes where necessary. The app was built starting from an initial create-react-app environment using Material UI for UI components, a mixture of vanilla CSS and CSS-in-JS for custom styling, and Express.js for the server and routing.</p>

            <h3 className='mb-4'><strong>Version 4 (React with Tailwind)</strong></h3>
            <p className='mb-8'>The current version was rebuilt again in <time dateTime='2022-01'>January 2022</time> with React and TailwindCSS as this became my preferred CSS framework. Additionally, JavaScript was upgraded to TypeScript to improve code quality, testing and help me learn the language.</p>
        </Section>

        <Divider />

        <Screenshots images={images} />

        <Divider />
        
        <Pagination
            previousTitle='LucidLab'
            previousLink='/projects/lucidlab'
        />
        </>
    );
}
