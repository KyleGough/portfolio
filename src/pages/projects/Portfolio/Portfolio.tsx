import React, { useEffect } from 'react';

import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { ProjectHeader } from '../../../components/ProjectHeader';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import { getProjectData } from '../../../data';
import images from './images.json';

const project = getProjectData('portfolio');

export const Portfolio: React.FC = () => {
  useEffect(() => {
    document.title = 'Portfolio - Personal Portfolio Website';
  }, []);

  return (
    <>
      <ProjectHeader project={project} />

      <Divider />

      <ImageFigure image={images[5]} />

      <Divider />

      <Section>
        <h2 className="project-header">Versions</h2>
        <h3 className="mb-4">
          <strong>Version 1 (HTML)</strong>
        </h3>
        <ul className="project-list mb-16">
          <li>
            The initial version created in{' '}
            <time dateTime="2016-12">December 2016</time>.
          </li>
          <li>Simple static site created with HTML, Sass and Bootstrap.</li>
          <li>
            As additional project pages were added the CSS framework was ported
            to Materialize.
          </li>
          <li>This version was never made public.</li>
        </ul>

        <h3 className="mb-4">
          <strong>Version 2 (PHP)</strong>
        </h3>
        <ul className="project-list mb-16">
          <li>
            Rebuilt in PHP to aid code reuse, maintainability, and expand
            functionality.
          </li>
          <li>
            Reusable components such as pagination, navigation bars, footers,
            and skill progress bars.
          </li>
          <li>
            However, these components were not easily maintainable or readable.
          </li>
          <li>
            <a
              className="text-link hover:text-link-hover focus:text-link-hover"
              href="https://github.com/KyleGough/portfolio-php"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub Repository
            </a>
          </li>
        </ul>

        <h3 className="mb-4">
          <strong>Version 3 (React with Materialize)</strong>
        </h3>
        <ul className="project-list mb-16">
          <li>
            Rebuilt in <time dateTime="2020-05">May 2020</time> with React due
            to its benefits and ease of use over PHP.
          </li>
          <li>
            Effort taken to replicate most of the previous styling but made
            specific changes where necessary.
          </li>
          <li>
            The app was built starting from an initial create-react-app
            environment.
          </li>
          <li>Materialize for UI components.</li>
          <li>Mixture of vanilla CSS and CSS-in-JS for custom styling.</li>
          <li>
            <a
              className="text-link hover:text-link-hover focus:text-link-hover"
              href="https://github.com/KyleGough/portfolio-react"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub Repository
            </a>
          </li>
        </ul>

        <h3 className="mb-4">
          <strong>Version 4 (React with Tailwind)</strong>
        </h3>
        <ul className="project-list mb-8">
          <li>
            Rebuilt in <time dateTime="2022-01">January 2022</time> with React
            and TailwindCSS - my preferred CSS framework.
          </li>
          <li>
            JavaScript was upgraded to TypeScript to improve code quality,
            testing and help me learn the language.
          </li>
          <li>Created custom CSS animations.</li>
          <li>
            Additional care and consideration into responsiveness, semantic
            HTML, and SEO.
          </li>
          <li>
            <a
              className="text-link hover:text-link-hover focus:text-link-hover"
              href="https://github.com/KyleGough/portfolio"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub Repository
            </a>
          </li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="LucidLab"
        previousLink="/projects/lucidlab"
        nextTitle="Bookmark Labeller"
        nextLink="/projects/bookmark-labeller"
      />
    </>
  );
};
