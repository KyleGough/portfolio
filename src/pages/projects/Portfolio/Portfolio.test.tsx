import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { Portfolio } from './Portfolio';

describe('Portfolio project page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <Portfolio />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
