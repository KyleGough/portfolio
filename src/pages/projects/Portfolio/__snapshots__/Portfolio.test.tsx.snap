// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio project page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      Portfolio
    </h1>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2016-12"
      >
        December 2016
      </time>
       
      -
       
      <time
        datetime="2022-10"
      >
        October 2022
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Personal portfolio website created to showcase my projects and technical skills. First created in 2016 as a simple static HTML website, the website has experienced multiple improvements over the years including migration to PHP, then to React with Materialize, and most recently to React with Tailwind.
    </p>
    <p
      class="mb-4"
    >
      <span>
        Explore this project on 
      </span>
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="https://github.com/KyleGough/portfolio"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        React
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        TypeScript
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        JavaScript
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Tailwind
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        HTML
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        CSS
      </div>
      <div
        class="text-disabled bg-chip-old text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        PHP
      </div>
      <div
        class="text-disabled bg-chip-old text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Materialize
      </div>
      <div
        class="text-disabled bg-chip-old text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Bootstrap
      </div>
      <div
        class="text-disabled bg-chip-old text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Sass
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Portfolio project page lower version 4"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/portfolio-projects-v4b.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Portfolio project page lower version 4
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Versions
    </h2>
    <h3
      class="mb-4"
    >
      <strong>
        Version 1 (HTML)
      </strong>
    </h3>
    <ul
      class="project-list mb-16"
    >
      <li>
        The initial version created in
         
        <time
          datetime="2016-12"
        >
          December 2016
        </time>
        .
      </li>
      <li>
        Simple static site created with HTML, Sass and Bootstrap.
      </li>
      <li>
        As additional project pages were added the CSS framework was ported to Materialize.
      </li>
      <li>
        This version was never made public.
      </li>
    </ul>
    <h3
      class="mb-4"
    >
      <strong>
        Version 2 (PHP)
      </strong>
    </h3>
    <ul
      class="project-list mb-16"
    >
      <li>
        Rebuilt in PHP to aid code reuse, maintainability, and expand functionality.
      </li>
      <li>
        Reusable components such as pagination, navigation bars, footers, and skill progress bars.
      </li>
      <li>
        However, these components were not easily maintainable or readable.
      </li>
      <li>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/portfolio-php"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub Repository
        </a>
      </li>
    </ul>
    <h3
      class="mb-4"
    >
      <strong>
        Version 3 (React with Materialize)
      </strong>
    </h3>
    <ul
      class="project-list mb-16"
    >
      <li>
        Rebuilt in 
        <time
          datetime="2020-05"
        >
          May 2020
        </time>
         with React due to its benefits and ease of use over PHP.
      </li>
      <li>
        Effort taken to replicate most of the previous styling but made specific changes where necessary.
      </li>
      <li>
        The app was built starting from an initial create-react-app environment.
      </li>
      <li>
        Materialize for UI components.
      </li>
      <li>
        Mixture of vanilla CSS and CSS-in-JS for custom styling.
      </li>
      <li>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/portfolio-react"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub Repository
        </a>
      </li>
    </ul>
    <h3
      class="mb-4"
    >
      <strong>
        Version 4 (React with Tailwind)
      </strong>
    </h3>
    <ul
      class="project-list mb-8"
    >
      <li>
        Rebuilt in 
        <time
          datetime="2022-01"
        >
          January 2022
        </time>
         with React and TailwindCSS - my preferred CSS framework.
      </li>
      <li>
        JavaScript was upgraded to TypeScript to improve code quality, testing and help me learn the language.
      </li>
      <li>
        Created custom CSS animations.
      </li>
      <li>
        Additional care and consideration into responsiveness, semantic HTML, and SEO.
      </li>
      <li>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/portfolio"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub Repository
        </a>
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio project page version 1"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v1a.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio project page version 1
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio about page version 1"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v1b.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio about page version 1
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio project page version 2"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v2.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio project page version 2
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio project page version 3"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v3.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio project page version 3
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio project page upper version 4"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v4a.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio project page upper version 4
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Portfolio project page lower version 4"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/portfolio-projects-v4b.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Portfolio project page lower version 4
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/lucidlab"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            LucidLab
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-disabled text-disabled cursor-not-allowed"
        href="/"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            -
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
